src/test/java/com/feedzai/commons/sql/abstraction/engine/impl/abs/AbstractEngineSchemaTest.java

```diff
@@ -420,6 +425,38 @@ public void testInsertBatchRandomValuesDoNoWorkInBinaryDoubleColumn() throws Exc
         testInsertSpecialValuesByBatch("randomString");
     }
 
+    /**
+     * Tests if the timeout setting is properly set in the connection socket to the database.
+     * Note: the DB2 and H2 databases don't support setting the connection socket timeout
+     *
+     * @throws DatabaseFactoryException
+     * @throws ClassNotFoundException
+     * @throws InterruptedException
+     * @throws RecoveryException
+     * @throws RetryLimitExceededException
+     * @throws SQLException
+     */
+    @Test
+    public void testTimeout() throws DatabaseFactoryException, ClassNotFoundException, InterruptedException, RecoveryException, RetryLimitExceededException, SQLException {
+        final int socketTimeout = 60;// in seconds
+        final int loginTimeout = 30;// in seconds
+        final Properties properties = new Properties() {
+            {
+                setProperty(JDBC, config.jdbc);
+                setProperty(USERNAME, config.username);
+                setProperty(PASSWORD, config.password);
+                setProperty(ENGINE, config.engine);
+                setProperty(SCHEMA_POLICY, "create");
+                setProperty(SOCKET_TIMEOUT, Integer.toString(socketTimeout));
+                setProperty(LOGIN_TIMEOUT, Integer.toString(loginTimeout));
+            }
+        };
+
+        final DatabaseEngine de = DatabaseFactory.getConnection(properties);
+        final int connectionTimeoutInMs = de.getConnection().getNetworkTimeout();
+        assertEquals("timeout in db connection ok?", socketTimeout * 1000, connectionTimeoutInMs);
```

Comment:
Error message could be more specific: "Is the timeout of the DB connection the expected?"

Reply:
Yes, that's better. Thanks.